build-utils/sentry-instrumentation.ts

```diff
@@ -1,149 +1,68 @@
       return;
     }
 
-    const sentry = require('@sentry/node') as typeof Sentry;
-    const {nodeProfilingIntegration} = require('@sentry/profiling-node');
-
-    sentry.init({
+    this.Sentry.init({
       dsn: 'https://3d282d186d924374800aa47006227ce9@sentry.io/2053674',
       environment: IS_CI ? 'ci' : 'local',
       tracesSampleRate: 1.0,
-      integrations: [nodeProfilingIntegration()],
-      profilesSampler: ({transactionContext}) => {
-        if (transactionContext.name === INCREMENTAL_BUILD_TXN) {
-          return 0;
-        }
-        return 1;
-      },
-      _experiments: {
-        // 5 minutes should be plenty
-        maxProfileDurationMs: 5 * 60 * 1000,
-      },
     });
 
-    if (IS_CI) {
-      sentry.setTag('branch', GITHUB_REF);
-    }
-
-    const cpus = os.cpus();
-    sentry.setTag('platform', os.platform());
-    sentry.setTag('arch', os.arch());
-    sentry.setTag(
-      'cpu',
-      cpus?.length ? `${cpus[0]!.model} (cores: ${cpus.length})}` : 'N/A'
-    );
-
-    this.Sentry = sentry;
+    this.withCITags();
+    this.withOSPlatformTags();
 
-    this.span = sentry.startInactiveSpan({
+    this.spans['initial-build'] = this.Sentry.startInactiveSpan({
       op: 'webpack-build',
-      name: INITIAL_BUILD_TXN,
+      name: 'initial-build',
     });
   }
 
-  /**
-   * Measures the file sizes of assets emitted from the entrypoints
-   */
-  measureAssetSizes(compilation: webpack.Compilation) {
-    if (!SENTRY_WEBPACK_WEBHOOK_SECRET) {
-      return;
+  withCITags() {
+    if (IS_CI) {
+      this.Sentry?.setTag('branch', GITHUB_REF);
     }
+  }
 
-    [...compilation.entrypoints].forEach(([entrypointName, entry]) =>
-      entry.chunks.forEach(chunk =>
-        Array.from(chunk.files)
-          .filter(assetName => !assetName.endsWith('.map'))
-          .forEach(assetName => {
-            const asset = compilation.assets[assetName];
-            const size = asset!.size();
-            const file = assetName;
-            const body = JSON.stringify({
-              file,
-              entrypointName,
-              size,
-              commit: GITHUB_SHA,
-              environment: IS_CI ? 'ci' : '',
-              node_env: NODE_ENV,
-            });
-
-            const req = https.request({
-              host: 'product-eng-webhooks-vmrqv3f7nq-uw.a.run.app',
-              path: '/metrics/webpack/webhook',
-              method: 'POST',
-              headers: {
-                'Content-Type': 'application/json',
-                'Content-Length': Buffer.byteLength(body),
-                'x-webpack-signature': createSignature(
-                  SENTRY_WEBPACK_WEBHOOK_SECRET,
-                  body
-                ),
-              },
-            });
-            req.end(body);
-          })
-      )
+  withOSPlatformTags() {
+    this.Sentry?.setTag('platform', os.platform());
+    this.Sentry?.setTag('arch', os.arch());
+
+    const cpus = os.cpus();
+    this.Sentry?.setTag(
```

Comment:
is optional chaining needed here?

Reply:
Not needed, going to remove it. Good case for type aware linting, this would scream with no unnecessary condition